pyiron_base/generic/hdfio.py

```diff
@@ -291,40 +291,50 @@ def copy_to(self, destination, file_name=None, maintain_name=True):
             FileHDFio: FileHDFio object pointing to a file which now contains the same content as file of the current
                        FileHDFio object.
         """
+        def _internal_copy(source, target, dest, h5_path):
+            if dest.h5_path[0] == "/":
+                dest_path = dest.h5_path[1:]
+            else:
+                dest_path = dest.h5_path
+            if maintain_name:
+                try:
+                    target.create_group(dest_path)
+                except ValueError:
+                    pass
+            if dest.is_root:
+                if source == target:
+                    source.copy(h5_path, "/")
+                else:
+                    source.copy(h5_path, target)
+            else:
+                if maintain_name:
+                    source.copy(h5_path, target[dest_path])
+                else:
+                    group_name_old = h5_path.split("/")[-1]
+                    try:
+                        target.create_group("/tmp")
+                    except ValueError:
+                        pass
+                    source.copy(h5_path, target["/tmp"])
+                    try:
+                        target.move("/tmp/" + group_name_old, dest_path)
+                    except ValueError:
+                        del target[dest_path]
+                        target.move("/tmp/" + group_name_old, dest_path)
+                    del target["/tmp"]
+
```

Comment:
In general I found this section really hard to read -- we get down to four levels of nesting in `if/try`. I'll open it in pycharm and see if I can see a more transparent way to structure it, but I don't want to hold up the review for this and we can always improve it in a future PR.

Reply:
Update: I actually have a meeting coming up pretty fast and probably won't get to this on the scale of hours. Unless you're suddenly convicted it can be much more readable, let's shelve it for now.